logviewer/src/main/java/com/nodesagency/logviewer/screens/logdetails/LogDetailsFragment.kt

```diff
@@ -42,18 +41,39 @@ internal class LogDetailsFragment : DialogFragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        logRepository = Logger.getRepository()
+        val logRepository = Logger.getRepository()
+        viewModel = ViewModelProviders
+            .of(this, LogDetailsViewModelFactory(logRepository, logEntryId))
+            .get(LogDetailsViewModel::class.java)
+
+
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         return inflater.inflate(R.layout.fragment_log_details, container, false)
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+
+        viewModel.logEntryLiveData.observe(this, Observer { logDetails ->
+            logDetailsView.setLogDetails(logDetails)
+
+            logDetailsShareBtn.setOnClickListener {
+                val sendIntent: Intent = Intent().apply {
+                    action = Intent.ACTION_SEND
+                    putExtra(Intent.EXTRA_TEXT, logDetails.toShareMessage())
+                    type = "text/plain"
+                }
+                startActivity(Intent.createChooser(sendIntent, "Send log"))
```

Comment:
Could you move this string to `strings.xml`, please? It would be good to keep this consistent for potential localizations later.

Reply:
Fixed :)